src/components/SkeletonLoading.vue

```diff
@@ -0,0 +1,181 @@
+<template>
+	<div class="placeholder-main placeholder-main-text">
+		<!-- Placeholder animation -->
+		<template v-for="(suffix, gradientIndex) in ['-regular', '-reverse']">
+			<svg :key="'gradient' + suffix" :class="'placeholder-gradient placeholder-gradient' + suffix">
+				<defs>
+					<linearGradient :id="'placeholder-gradient' + suffix">
+						<stop offset="0%" :stop-color="(gradientIndex === 0) ? colorPlaceholderLight : colorPlaceholderDark" />
+						<stop offset="100%" :stop-color="(gradientIndex === 0) ? colorPlaceholderDark : colorPlaceholderLight" />
+					</linearGradient>
+				</defs>
+			</svg>
+
+			<ul :key="'list' + suffix" :class="'placeholder-list placeholder-list' + suffix">
+				<li v-for="(width, index) in placeholderData" :key="'placeholder' + suffix + index">
+					<svg class="text-placeholder"
+						xmlns="http://www.w3.org/2000/svg"
+						:fill="'url(#placeholder-gradient' + suffix + ')'">
+						<rect class="text-placeholder-line-one" :style="textLinesWidth[0]" />
+						<rect class="text-placeholder-line-two" :style="textLinesWidth[1]" />
+						<rect class="text-placeholder-line-three" :style="textLinesWidth[2]" />
+						<rect class="text-placeholder-line-four" :style="width" />
+					</svg>
+				</li>
+			</ul>
+		</template>
+	</div>
+</template>
+
+<script>
+const bodyStyles = window.getComputedStyle(document.body)
+const colorPlaceholderDark = bodyStyles.getPropertyValue('--color-placeholder-dark')
+const colorPlaceholderLight = bodyStyles.getPropertyValue('--color-placeholder-light')
+
+export default {
+	name: 'SkeletonLoading',
+
+	props: {
+		count: {
+			type: Number,
+			default: 5,
+		},
+	},
+
+	setup() {
+		return {
+			colorPlaceholderDark,
+			colorPlaceholderLight,
+		}
+	},
+
+	computed: {
+		placeholderData() {
+			const data = []
+			for (let i = 0; i < this.count; i++) {
+				// generate random widths
+				data.push('width: ' + (Math.floor(Math.random() * 40) + 50) + '%')
+			}
+			return data
+		},
+		textLinesWidth() {
+			return [
+				'width: ' + (Math.floor(Math.random() * 50) + 60) + '%',
+				'width: ' + (Math.floor(Math.random() * 50) + 60) + '%',
+				'width: ' + (Math.floor(Math.random() * 50) + 60) + '%',
+			]
+		},
+	},
+}
+</script>
+
+<style lang="scss" scoped>
+$clickable-area: 44px;
+$margin: 8px;
+$messages-list-max-width: 670px;
+
+.placeholder-main {
+	max-width: $messages-list-max-width;
+	position: relative;
+	margin-bottom: auto;
```

Comment:
I noticed there is some flickering during loading when the editor div appears the skeleton is hidden behind it. Adding a `z-index: 1;` here seems to fix this.